extensions/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/RestEasyCORSTestCase.java

```diff
@@ -17,6 +17,21 @@
                     .addClasses(RootResource.class)
                     .addAsResource("cors-config.properties", "application.properties"));
 
+    @Test
+    public void testCORSPreflightRootResource() {
+        String origin = "http://custom.origin.quarkus";
+        String methods = "GET,POST";
+        String headers = "X-Custom";
+        RestAssured.given()
+                .header("Origin", origin)
+                .header("Access-Control-Request-Method", methods)
+                .header("Access-Control-Request-Headers", headers)
```

Comment:
Quick question: these 2 headers are always defined?

Reply:
According to the spec they should be defined. see https://fetch.spec.whatwg.org/#http-requests

Reply:
See also: https://developer.mozilla.org/en-US/docs/Glossary/preflight_request

Reply:
I did it this way because I wanted this to be independent of the cors config and it is based on a standard. The other alternative would be making the ResteasyFilter aware of the HttpConfig and check if CORS was enable... but I think it is too invasive and we would start having CORS logic all around the web extensions.